prov/rxm/src/rxm_conn.c

```diff
@@ -484,6 +489,27 @@ void rxm_cmap_process_conn_notify(struct rxm_cmap *cmap,
 	rxm_conn_connected_handler(handle);
 }
 
+static inline void rxm_cmap_progress_wait_conn_quota_queue(struct rxm_cmap *cmap)
+{
+	int ret;
+	struct rxm_cmap_handle *next_handle;
+
+	slist_remove_head_container(&cmap->conn_wait_quota_list,
+				    struct rxm_cmap_handle,
+				    next_handle, wait_conn_quota_entry);
+	if (next_handle) {
+		slist_entry_init(&next_handle->wait_conn_quota_entry);
+
+		ret = rxm_cmap_handle_connect(cmap, next_handle->fi_addr, next_handle);
+		if (ret && (ret != -FI_EAGAIN) && (ret != RXM_CMAP_SHUTDOWN)) {
+			FI_WARN(&rxm_prov, FI_LOG_EP_CTRL,
+				"Handle connect fails with %d. Try again\n", ret);
+			slist_insert_tail(&next_handle->wait_conn_quota_entry,
+					  &cmap->conn_wait_quota_list);
+		}
```

Comment:
Where does the cmap_handle go if there's an actual error or the cmap is shutting down?

Reply:
There should probably be some way to report the error or handle it to avoid an endless loop.

Reply:
@shefty Now the code deletes a connection for which we receive an error. So, the error is propagated to a caller when fi_send*() is failed, because rxm_conn for this fi_addr_t is NULL.